includes/shell/executor/builtins.hpp

```diff
@@ -0,0 +1,47 @@
+
+#pragma once
+
+#include "shell/ast.hpp"
+#include "executor.hpp"
+#include <functional>
+
+namespace shell
+{
+	/*!
+	 * @brief Singleton class for builtins.
+	*/
+	class Builtin
```

Comment:
I don't think this needs to be a singleton, it can just be a class with only static methods, the only reason this is currently a singleton currently is because of `std::map<string, builtinFunction> builtins;` but I think we can just statically declare that inside the static `getBuiltin` method instead.